tutorial/westeros/westeros_seasonality.ipynb

```diff
@@ -0,0 +1,601 @@
+    "scen.add_set('time', time_steps)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We can see the elements of the set\n",
+    "scen.set('time')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Defining a new temporal level\n",
+    "time_level = 'season'\n",
+    "scen.add_set('lvl_temporal', time_level)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next, the temporal hierarchy will be defined to map different levels of time with respect to \"year\", which is the parent temporal level for any \"time\". "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Adding temporal hierarchy\n",
+    "for t in time_steps:\n",
+    "    scen.add_set('map_temporal_hierarchy', [time_level, t, 'year'])\n",
+    "    \n",
+    "# We can see the content of the set\n",
+    "scen.set('map_temporal_hierarchy')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.3. Modifying parameters\n",
+    "In this Section, we modify some parameters based on the new time steps. In principle, we need to examine all parameters that have an index of \"time\" to see if we need to modify them or not."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# All parameters with at least one sub-annual time index\n",
+    "parameters = [p for p in scen.par_list() if 'time' in scen.idx_sets(p)]\n",
+    "\n",
+    "# Those parameters with \"time\" index that are not empty in our model\n",
+    "[p for p in parameters if not scen.par(p).empty]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Parameter \"duration_time\"\n",
+    "We start by modifying the parameter \"duration_time\", which shows the length of each subannual time step relative to the whole year:\n",
+    "\n",
+    "| Parameter | Index set| Explanation|\n",
+    "|-----------|----------| -----------|\n",
+    "| duration_time\t| time\t|duration of sub-annual time slices (relative to 1)| \n",
+    "\n",
+    "In our example, winter and summer are each half of the year."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Adding duration time\n",
+    "for t in time_steps:\n",
+    "    scen.add_par('duration_time', [t], 0.5, '-')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### A function for modifying parameters\n",
+    "In this stage, we introduce a function that helps us to modify the parameters after adding new time steps. This function called \"yearly_to_season\" does the following:\n",
```

Comment:
Make the general point clear here: “This is a common code pattern when modelling using MESSAGEix: writing re-usable code that helps modify existing parameter data to reflect some desired change in the reference energy system.” or something like that.

Reply:
Thanks for a nice explanation @khaeru.